src/Bridge/Symfony/Validator/Metadata/Property/Restriction/PropertySchemaChoiceRestriction.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Symfony\Validator\Metadata\Property\Restriction;
+
+use ApiPlatform\Core\Metadata\Property\PropertyMetadata;
+use Symfony\Component\PropertyInfo\Type;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\Constraints\Choice;
+
+/**
+ * @author Tomas Norkūnas <norkunas.tom@gmail.com>
+ */
+class PropertySchemaChoiceRestriction implements PropertySchemaRestrictionMetadataInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function create(Constraint $constraint, PropertyMetadata $propertyMetadata): array
+    {
+        if (!\in_array($builtinType = $propertyMetadata->getType()->getBuiltinType(), [Type::BUILTIN_TYPE_STRING, Type::BUILTIN_TYPE_INT, Type::BUILTIN_TYPE_FLOAT], true)) {
+            return [];
+        }
+
+        \assert($constraint instanceof Choice);
```

Comment:
We usally don't use `assert` in API Platform.

Reply:
But then PHPStan complains:
```
  33     Access to an undefined property                                                                 
         Symfony\Component\Validator\Constraint::$callback.                                              
  35     Access to an undefined property                                                                 
         Symfony\Component\Validator\Constraint::$choices.                                               
  45     Access to an undefined property                                                                 
         Symfony\Component\Validator\Constraint::$multiple.                                              
  54     Access to an undefined property                                                                 
         Symfony\Component\Validator\Constraint::$min.                                                   
  58     Access to an undefined property                                                                 
         Symfony\Component\Validator\Constraint::$max.       
```

Reply:
You just need to add the right PHPDoc.

Reply:
okay, let's see how it goes